unittests/internaltests.py

```diff
@@ -674,6 +674,36 @@ def _call_pkgbin(self, args, env=None):
                         for lib in ('pthread', 'm', 'c', 'dl', 'rt'):
                             self.assertNotIn(f'lib{lib}.a', link_arg, msg=link_args)
 
+    def test_program_version(self):
+        if is_windows():
+            return
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            script_path = Path(tmpdir) / 'script'
+            script_path.write_text('#!/bin/sh\necho "$1"\n', encoding='utf-8')
```

Comment:
I think this test case would fail on Windows systems if, unlike the Azure CI environment, there is no mingw shell / Git Bash preinstalled.

Instead it returns early on Windows, which feels vaguely surprising? We could write out an equivalently small python script here instead.

Reply:
Yeah, my initial version of this test was a bit more complicated until I got the idea that I can override the command, including its arguments. So that's why I didn't opt for a Python script initially. But now in the simplified version it's easy enough to do, so I've adapted it accordingly.